flink-runtime/src/test/java/org/apache/flink/runtime/iterative/event/EventWithAggregatorsTest.java

```diff
@@ -123,6 +169,31 @@ private IterationEventWithAggregators pipeThroughSerialization(IterationEventWit
 			return null;
 		}
 	}
+
+	private ClockTaskEvent pipeThroughSerialization2(ClockTaskEvent event) {
+		try {
+			ByteArrayOutputStream baos = new ByteArrayOutputStream();
+			DataOutputStream out = new DataOutputStream(baos);
+			event.write(new OutputViewDataOutputStreamWrapper(out));
+			out.flush();
+
+			byte[] data = baos.toByteArray();
+			out.close();
+			baos.close();
+
+			DataInputStream in = new DataInputStream(new ByteArrayInputStream(data));
+			ClockTaskEvent newEvent = event.getClass().newInstance();
+			newEvent.read(new InputViewDataInputStreamWrapper(in));
+			in.close();
+
+			return newEvent;
+		} catch (Exception e) {
+			System.err.println(e.getMessage());
+			e.printStackTrace();
+			Assert.fail("Test threw an exception: " + e.getMessage());
+			return null;
```

Comment:
I actually reproduced the same as in ```testSerializationOfEventWithAggregateValues```. What you say makes sense in both cases though.